opentelemetry-api/src/opentelemetry/util/__init__.py

```diff
@@ -33,18 +34,44 @@ def time_ns() -> int:
         return int(time.time() * 1e9)
 
 
-def _load_provider(provider: str) -> Union["TracerProvider", "MeterProvider"]:  # type: ignore
+def _load_provider(
+    provider: str,
+) -> Union["TracerProvider", "MeterProvider"]:  # type: ignore
     try:
         return next(  # type: ignore
             iter_entry_points(
                 "opentelemetry_{}".format(provider),
-                name=getattr(  # type: ignore
-                    Configuration(), provider, "default_{}".format(provider),  # type: ignore
+                name=getattr(
+                    Configuration(),  # type: ignore
+                    provider,
+                    "default_{}".format(provider),
                 ),
             )
         ).load()()
     except Exception:  # pylint: disable=broad-except
-        logger.error(
-            "Failed to load configured provider %s", provider,
-        )
+        logger.error("Failed to load configured provider %s", provider)
         raise
+
+
+# Pattern for matching the 'https://', 'http://', 'ftp://' part.
+URL_PATTERN = "^(https?|ftp):\\/\\/"
+
+
+def disable_tracing_path(url: str, excluded_paths: Sequence[str]) -> bool:
```

Comment:
I think this is worth benchmarking, but if the list of regex become long here, then you may end up with a situation where the cost of running regex queries against the path will significant overhead and will need to be memoized.

I wonder if an lrucache might be a good idea here. Although to @ocelotl's point, you're probably getting a huge benefit by joining the regexes to a single query to skip a lot of unneeded matching.

Reply:
Fixed regex.